docroot/modules/custom/foia_webform/src/FoiaSubmissionProcessingFactory.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Drupal\foia_webform\Plugin\QueueWorker;
+
+use Drupal\node\Entity\Node;
+
+/**
+ * Class FoiaSubmissionProcessingFactory.
+ *
+ * @package Drupal\foia_webform\Plugin\QueueWorker
+ */
+class FoiaSubmissionProcessingFactory {
+
+  /**
+   * Check the submission format and return the service.
+   *
+   * @param \Drupal\node\Entity\Node $agencyComponent
+   *   The Agency Component node object.
+   *
+   * @return \Drupal\foia_webform\FoiaSubmissionServiceInterface
+   *   Returns a Fo
+   */
+  public function get(Node $agencyComponent) {
+    $submissionPreference = $agencyComponent->get('field_portal_submission_format');
+
+    switch ($submissionPreference) {
+      case 'api':
+        $serviceName = 'foia_webform.foia_submission_api_service';
+        break;
+
+      case 'email':
+        $serviceName = 'foia_webform.foia_submission_email_service';
+        break;
+
+      default:
+        $serviceName = 'foia_webform.foia_submission_email_service';
```

Comment:
Same comment as above, let's leave off email-related features for now. We can add it in with 122.